sql/src/main/java/io/crate/metadata/doc/DocTableInfoBuilder.java

```diff
@@ -161,13 +163,21 @@ private DocIndexMetaData buildDocIndexMetaDataFromTemplate(String index, String
         return partitions;
     }
 
+    private Collection<Reference> buildSysColumns() {
+        ArrayList<Reference> result = new ArrayList<>();
+        DocSysColumns.forTable(this.ident, (c,r) -> result.add(r));
+        return result;
+    }
+
     public DocTableInfo build() {
         DocIndexMetaData md = docIndexMetaData();
         List<PartitionName> partitions = buildPartitions(md);
+        Collection<Reference> sysColumns = buildSysColumns();
 
         return new DocTableInfo(
             ident,
             md.columns(),
+            sysColumns,
```

Comment:
```suggestion
            DocSysColumns.forTable(ident),
```

Reply:
Or maybe we can remove this from the `DocTableInfo` altogether, and instead create it ad-hoc in the `DocTableRelation` ? Not sure which way is better. Higher memory footprint vs. more ad-hoc allocations